app/shop.py

```diff
@@ -0,0 +1,40 @@
+from __future__ import annotations
+import datetime
+
+
+class Shop:
+    def __init__(self, shop: dict) -> None:
+        self.name: str = shop.get("name")
+        self.location: list[int] = shop.get("location")
+        self.products: dict[str, float | int] = shop.get("products")
+
+    @staticmethod
+    def format_value(value: float) -> str:
+        return str(value).rstrip("0").rstrip(".")
+
+    def print_receipt(
+            self,
+            customer_name: str,
+            customer_products: dict
+    ) -> None:
+        date = datetime.datetime.now().strftime("%d/%m/%Y %H:%M:%S")
+        receipt: str = ""
+        receipt += f"Date: {date}\n"
+        receipt += f"Thanks, {customer_name}, for your purchase!\n"
+        receipt += "You have bought:\n"
+        total_cost = 0
+
+        for product, count in customer_products.items():
+            price = count * self.products.get(product, 0)

```

Comment:
Ensure that the `self.products` dictionary contains all products listed in `customer_products`. If a product is missing, the default price of 0 will be used, which might not be the desired behavior. Consider adding a check or warning for missing products.